nntrainer/layers/fc_layer.cpp

```diff
@@ -196,27 +246,54 @@ void FullyConnectedLayer::calcDerivative(RunLayerContext &context) {
 }
 
 void FullyConnectedLayer::calcGradient(RunLayerContext &context) {
-  Tensor &djdw = context.getWeightGrad(weight_idx[FCParams::weight]);
 
-  const Tensor &derivative_ = context.getIncomingDerivative(SINGLE_INOUT_IDX);
-  Tensor &input_ = context.getInput(SINGLE_INOUT_IDX);
+  // (baseline) calcGradient
+  if (std::get<props::LoraRank>(fc_props).empty()) {
+    Tensor &djdw = context.getWeightGrad(weight_idx[FCParams::weight]);
 
-  if (auto &disable_bias = std::get<props::DisableBias>(*layer_impl_props);
-      disable_bias.empty() || disable_bias.get() == false) {
-    Tensor &djdb = context.getWeightGrad(weight_idx[FCParams::bias]);
-
-    if (context.isGradientFirstAccess(weight_idx[FCParams::bias])) {
-      derivative_.sum({0, 1, 2}, djdb);
-    } else {
-      /// @todo optimize below by adding beta to Tensor::sum
-      Tensor t = derivative_.sum({0, 1, 2});
-      djdb.add_i(t);
+    const Tensor &derivative_ = context.getIncomingDerivative(SINGLE_INOUT_IDX);
+    Tensor &input_ = context.getInput(SINGLE_INOUT_IDX);
+
+    if (auto &disable_bias = std::get<props::DisableBias>(*layer_impl_props);
+        disable_bias.empty() || disable_bias.get() == false) {
+      Tensor &djdb = context.getWeightGrad(weight_idx[FCParams::bias]);
+
+      if (context.isGradientFirstAccess(weight_idx[FCParams::bias])) {
+        derivative_.sum({0, 1, 2}, djdb);
+      } else {
+        /// @todo optimize below by adding beta to Tensor::sum
+        Tensor t = derivative_.sum({0, 1, 2});
+        djdb.add_i(t);
+      }
     }
-  }
 
-  input_.dot_deriv_wrt_2(
-    djdw, derivative_, false, false,
-    !context.isGradientFirstAccess(weight_idx[FCParams::weight]));
+    input_.dot_deriv_wrt_2(
+      djdw, derivative_, false, false,
+      !context.isGradientFirstAccess(weight_idx[FCParams::weight]));
+  } else {
+    // (LoRA) calcGradient
+    Tensor &djdla = context.getWeightGrad(lora_idx[LORAParams::loraA]);
+    Tensor &djdlb = context.getWeightGrad(lora_idx[LORAParams::loraB]);
+    Tensor &djdlora_w = context.getTensorGrad(lora_idx[LORAParams::loraW]);
+
+    const Tensor &derivative_ = context.getIncomingDerivative(SINGLE_INOUT_IDX);
+    Tensor &input_ = context.getInput(SINGLE_INOUT_IDX);
+    Tensor &lora_A = context.getWeight(lora_idx[LORAParams::loraA]);
+    Tensor &lora_B = context.getWeight(lora_idx[LORAParams::loraB]);
+
+    // (cf) forward
```

Comment:
Is this a typo in FC?

Reply:
It meant 'confer (refer)'. It looks redundant,  so I removed it :)